src/frontend/screens/ObservationEdit/SaveButton.tsx

```diff
@@ -55,25 +56,41 @@ const m = defineMessages({
 const MINIMUM_ACCURACY = 10;
 const log = debug('SaveButton');
 
-export const SaveButton = ({observationId}: {observationId?: string}) => {
+export const SaveButton = ({
+  observationId,
+  openErrorModal,
+}: {
+  observationId?: string;
+  openErrorModal?: () => void;
+}) => {
   const value = usePersistedDraftObservation(store => store.value);
   const photos = usePersistedDraftObservation(store => store.photos);
-  const {clearDraft} = useDraftObservation();
   const {formatMessage: t} = useIntl();
   const navigation = useNavigationFromRoot();
   const createObservationMutation = useCreateObservation();
   const editObservationMutation = useEditObservation();
 
   function createObservation() {
     if (!value) throw new Error('no observation saved in persisted state ');
-    createObservationMutation.mutate({value, photos});
-    navigation.navigate('Home', {screen: 'Map'});
+    createObservationMutation.mutate(
+      {value},
+      {
+        onError: () => {
+          if (openErrorModal) openErrorModal();
+        },
+        onSuccess: () => {
+          navigation.navigate('Home', {screen: 'Map'});
+        },
+      },
+    );
   }
 
   function editObservation() {
     if (!value) throw new Error('no observation saved in persisted state ');
+    if (!observationId) throw new Error('Need an observation Id to edit');
     if (!('versionId' in value))
       throw new Error('Cannot update a unsaved observation (must create one)');
+    // @ts-expect-error
     editObservationMutation.mutate({id: observationId, value});
```

Comment:
Again, sorry, I put this `// @ts-expect-error` here because I was going to deal with it later and it was out of scope of this PR.

Reply:
Yeah no worries! just thought it was worth keeping in the back of the mind for the mapeo core work in case we needed to make adjustments there based on this